jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitRunner.kt

```diff
@@ -51,14 +52,24 @@ class SamInitRunner(
     }
 
     companion object {
+        private val expectedSamVersion = SemVer.parseFromText("0.6.0") ?: throw RuntimeException("SemVer parse error")
         fun validate(path: String): String? {
             val commandLine = GeneralCommandLine(path).withParameters("--version")
-            return try {
+            try {
                 val process = CapturingProcessHandler(commandLine).runProcess()
-                if (process.exitCode != 0) process.stderr else null
+                if (process.exitCode != 0) {
+                    return process.stderr
+                }
+                val samVersionLine = process.stdoutLines.first()
+                val parsedSemVer = SemVer.parseFromText(samVersionLine.split(" ").last())
```

Comment:
It's better to move the version parse logic to a separate method that we can have a unit test for it.